### PR TITLE
Initial Service Connection Manager

### PR DESCRIPTION
**Pull request by peterschwarz**
This initial implementation currently only accepts incoming connections and stores them in state running on a background thread.  

Future PRs will include:
- facilitating an exchange of authorization-like messages
- Multiple thread clients (analogous to the ConnectionManager's Connector struct)
- connecting out to services

### CODE REVIEW

File: libsplinter/Cargo.toml [hunk 1]
```diff
@@ -105,6 +105,7 @@ experimental = [
     "matrix",
     "network-peer-manager",
     "network-ref-map",
+    "network-service-manager",
```
Comment (agunde406) [inline libsplinter/Cargo.toml]: Typo in commit message "network-servicee-manager" -> "network-service-manager"

File: libsplinter/src/service/network/mod.rs [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    Shutdown,
+}
+
+pub struct ServiceConnectionManager {
+    _connector: Connector,
+    sender: Sender<AgentMessage>,
+    join_handle: thread::JoinHandle<()>,
+}
+
+macro_rules! agent_msg {
```
Comment (agunde406) [inline libsplinter/src/service/network/mod.rs]: Can you add a comment about what this marco does and why it is required?

File: libsplinter/src/service/network/mod.rs [hunk 3]
```diff
@@ -11,3 +11,458 @@
...
+/// ```
+/// agent_msg!(
+///     self.sender,
+///     GetConnectionId {
+///         connection_id: connection_id.to_string(),
+///     }
+/// )
+/// ```
+///
+/// This removes the repated error messages that are specific to the senders in this exchange, but
```
Comment (agunde406) [inline libsplinter/src/service/network/mod.rs:124]: repated -> repeated

File: libsplinter/src/service/mod.rs [hunk 4]
```diff
@@ -37,6 +37,8 @@
 
 pub mod error;
 mod factory;
+#[cfg(feature = "network-service-manager")]
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libsplinter/src/service/mod.rs]: renamed to `"service-network"`

File: libsplinter/src/service/mod.rs [hunk 5]
```diff
@@ -38,7 +38,7 @@
 pub mod error;
 mod factory;
 #[cfg(feature = "network-service-manager")]
-mod network;
+pub mod network;
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libsplinter/src/service/mod.rs:41]: added.

File: libsplinter/src/service/network/mod.rs [hunk 6]
```diff
@@ -11,3 +11,579 @@
...
+use std::thread;
+
+use crate::network::connection_manager::{ConnectionManagerNotification, Connector};
+
+use self::error::ServiceConnectionAgentError;
+pub use self::error::ServiceConnectionError;
+
+/// Constructs new ServiceConnectionManager structs.
+///
+/// This builder is used to constructs a ServiceConnectionManager. At build time, this has
```
Comment (agunde406) [inline libsplinter/src/service/network/mod.rs:31]: used to constructs  -> used to construct

File: libsplinter/src/service/network/mod.rs (hunk 3 above)
Comment (peterschwarz) [reply libsplinter/src/service/network/mod.rs]: fixed

File: libsplinter/src/service/network/mod.rs (hunk 6 above)
Comment (peterschwarz) [reply libsplinter/src/service/network/mod.rs]: Removed the sentence - it was redundant.

**Pull request merged**